dracut.sh

```diff
@@ -2118,16 +2080,28 @@ fi
 
 if dracut_module_included "squash"; then
     dinfo "*** Squashing the files inside the initramfs ***"
+    declare squash_compress_arg
+    if [[ $compress_l ]]; then
+        if ! mksquashfs /dev/null "$DRACUT_TMPDIR"/.squash-test.img -no-progress -comp $compress_l &>/dev/null; then
+            dwarn "mksquashfs doesn't support compressor '$compress_l', failing back to default compressor."
+        else
+            squash_compress_arg="-comp $compress_l"
```

Comment:
Additionally, I would change this to 
```suggestion
            squash_compress_arg="$compress"
```
to use it later in the real call with `${squash_compress_arg:+-comp "$squash_compress_arg"}`